### PR TITLE
chore: migrate to dual publish

### PR DESCRIPTION
**Pull request by CatchMe2**
## Changes

This PR implements dual publish with tsup.
A new "exports" field was added to the package.json that replaces "main"/"types" fields.
tsconfig.json was updated to work as a typecheck process only, and a few strict params were added like verbatimModuleSyntax or "moduleDetection": "force".

It is a major release as it will break the codebase if any import is pointing to the submodules (@lokalise/node-core/dist/xxx)

## Checklist

- [x] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [x] I've updated the documentation, or no changes were necessary
- [x] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
+        "lib": ["ES2022"],
         "types": ["node", "vitest/globals"],
-        "strict": true,
+        "moduleDetection": "force",
+        "verbatimModuleSyntax": true,
         "skipLibCheck": true,
-        "moduleResolution": "node",
+        "esModuleInterop": true,
+        "resolveJsonModule": true,
+        "noEmit": true,
```
Comment (kibertoad) [inline tsconfig.json:12]: why do we use noEmit by default?

File: package.json [hunk 2]
```diff
@@ -11,20 +11,25 @@
...
+            "import": "./dist/index.js",
+            "default": "./dist/index.cjs"
+        }
+    },
     "publishConfig": {
         "access": "public"
     },
     "scripts": {
-        "build": "tsc",
+        "build": "tsup",
```
Comment (CarlosGamero) [inline package.json:27]: Just out of curiosity, and most likely I lack some knowledge 😓 so please bear with me.
AFAIK the current build system works with `cjs` and `esm` services and we are migrating our services to ESM, could make sense to wait until all services are migrated and then just go with ESM?

Sorry if the question is very stupid 🙏

File: tsconfig.json (hunk 1 above)
Comment (CatchMe2) [reply tsconfig.json:12]: typescript is used only for typechecking now, building is handled by tsup

File: package.json (hunk 2 above)
Comment (CatchMe2) [reply package.json:27]: Previously, we were building to `cjs` only, with tsup we are doing dual-build. In the dist there are .js (ESM) and .cjs (CJS) now. Besides that, the current approach prevents from accessing subpaths in imports.
Once all services are rewritten to ESM we can get rid of CJS entirely.

Comment (kibertoad) [reply package.json:27]: Considering that a lot of stuff depend on node-core, including some external libraries like message-queue-toolkit, which are used outside of just Lokalise, and forcing all of them to go ESM-only wouldn't be particularly nice. I'd say that dual publish makes sense until at least Node 24, which will finally support ESM imports in CJS world.

Comment (CatchMe2) [reply package.json:27]: ESM doesn't work without dual-publish. Based on local test
```
// tmp.mjs
import { removeDuplicates } from '@lokalise/node-core'

console.log(removeDuplicates([1, 1]))
```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in node_modules/@lokalise/node-core/package.json imported from polyglot-service/tmp.mjs

Comment (CatchMe2) [reply package.json:27]: @kibertoad yeah, we shouldn't go ESM only before Node 24 LTS

Comment (CarlosGamero) [reply package.json:27]: I see! thanks for explanation it makes a lot of sense 🙏

**Review by kibertoad — APPROVED** [review]

**Pull request merged**